### PR TITLE
[DI] Update CHANGELOG.md to include updates in the new API

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md [hunk 1]
```diff
@@ -2,4 +2,129 @@
 
 ## 1.0.0b1 (2023-11-17)
 
-- This is the first preview of the `azure-ai-documentIntelligence` package, targeting API version 2023-10-31-preview of the Document Intelligence service.
+This is the first preview of the `azure-ai-documentintelligence` package, targeting API version `2023-10-31-preview` of the Document Intelligence service(formerly known as Form Recognizer).
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:5]: To me, this content looks more appropriate for a migration guide than a changelog entry. We can always link to the migration guide from the changelog.

**Comment by YalinLi0312** [comment]
#assign-reviewer:laujan

**Comment by laujan** [comment]
@laujan

**Comment by YalinLi0312** [comment]
Hi @laujan , can you help to review this PR?

**Comment by laujan** [comment]
Hi @YalinLi0312, it looks good to me!

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+|DocumentAnalysisClient|DocumentIntelligenceClient|
+|DocumentModelAdministrationClient|DocumentIntelligenceAdministrationClient|
+
+### Base endpoint updates 
+Updates all REST API operation paths from `{endpoint}/formrecognizer` to `{endpoint}/documentintelligence`. SDK would handle this change automatically, users would not have to do additional work to support this.
+
+### Not backword compatible with azure-ai-formrecognizer
+`azure-ai-documentintelligence` is a new package, it is not compatible with the previous `azure-ai-formrecognizer` package without necessary changes to your code.
+
+### API shapes changes
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:101]:
```suggestion
### API changes
```

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+API shapes have been designed from scratch to support the new Client for the Document Intelligence service. Please refer to the [Readme](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/documentintelligence/azure-ai-documentintelligence/README.md) and [Samples](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/documentintelligence/azure-ai-documentintelligence/samples) for more understanding.
+
+### Fields changes in prebuilt-receipt model
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:104]: Nit:
```suggestion
### Field changes in prebuilt-receipt model
```

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            "currencyCode": "USD"
+        },
+    ...
+    },
+    "Tax": { "type": "currency", "valueCurrency": ... },
+    ...
+}
+```
+
+### Model Retirements/Deprecations.
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:136]:
```suggestion
### Model Retirements/Deprecations
```

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+- `"prebuilt-businessCard"` model is retired.
+
+- `"prebuilt-document"` model is retired, this model is essentially `"prebuilt-layout"` with `features="keyValuePairs"` specified. _(This is only supported as an optional feature for "prebuilt-layout" and "prebuilt-invoice".)_
+
+If you wish to still use these models, please rely on the older `azure-ai-formrecognizer` library through the older service API versions.
```
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:142]: For this section you should link to the DI supported models table and guide people to refer to that table to get an up to date view of supported models

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+client = DocumentIntelligenceClient(endpoint=endpoint, credential=AzureKeyCredential(key))
+poller = client.begin_analyze_document(
+    "prebuilt-layout", AnalyzeDocumentRequest(url_source=url), output_content_format=ContentFormat.MARKDOWN
+)
+result = poller.result()
+```
+
+### Query Fields
+
+We reintroduce query fields as a premium add-on feature. When the keyword argument `features` is specified, the service will further extract the values of the fields specified via the keyword argument `query_fields` to supplement any existing fields defined by the model as fallback.
```
[the remark this answers is not in the document]
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:47]: It was in one/some versions in old sdk, not the final GA. The "reintroduce" is talked in this service changes doc: https://mixedrealitywiki.com/pages/viewpage.action?pageId=245177517#DocumentIntelligence20231031preview-QueryFields

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+- `split: "perPage"`
+
+Each page is treated as a separate document. Each empty page is kept as its own document.
+
+## Breaking Changes
+
+### Clients names updates.
+
+|**2023-07-31**|**2023-10-31-preview**|
```
[the remark this answers is not in the document]
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:90]: Client names could change in different package versions. Using SDK package names with package version  or service API versions are both good for me. Which one do you think is better/less confusing?

File: sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md (hunk 5 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/MIGRATION_GUIDE.md:142]: Agree, I was also thinking about it but I didn't find the public link for DI supported models. It probably same like other DI links, not ready yet. We should link it here in future.

**Review by catalinaperalta — APPROVED** [review]
Approving, but left some comments for more improvements

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md [hunk 8]
```diff
@@ -2,4 +2,15 @@
...
+
+> Note: Form Recognizer has been rebranded to Document Intelligence.
+
+### Breaking Changes
+
+- Changed clients names from DocumentAnalysisClient and DocumentModelAdministrationClient in API version 2023-07-31 in `azure-ai-formrecognizer` to DocumentIntelligenceClient and DocumentIntelligenceAdministrationClient in API version 2023-10-31-preview in `azure-ai-documentintelligence`.
+- Changed all REST API operation paths from `{endpoint}/formrecognizer` to `{endpoint}/documentintelligence`.
+- Changed some currency-related fields in `prebuilt-receipt` model.
+- Retired model `prebuilt-businessCard` and `prebuilt-document`. `prebuilt-document` model is essentially `prebuilt-layout` with `features="keyValuePairs"` specified. _(This is only supported as an optional feature for "prebuilt-layout" and "prebuilt-invoice".)_
```
[the remark this answers is not in the document]
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:24]: That's the request from doc team, we all languages are adding CHANGELOG entries.

**Pull request merged**